Automattic/woocommerce-payments
# Migrate deprecated payment request button types "Branded" and "Custom"

Fixes #1991.

#### Changes proposed in this Pull Request

This PR introduces a migration that will run after the plugin is upgraded from a version prior to 2.6.0.

The migration will look at the `payment_request_button_type` field to see if its value is one of the now-deprecated values: `branded` or `custom`. If so, it will map it to a suitable fallback value, as per #1991.

There is a known issue with this PR that will cause the migration to only run if a wp-admin page is visited after the upgrade. Work on this is currently in progress in #2153.

Important: Values other than "Branded" or "Custom" will be left intact, including custom values that could be put in place by third-party plugins.









#### Testing instructions





* Manually update the `Version` header in `woocommerce-payments.php` to `Version: 2.6.0` to simulate the plugin being in version 2.6.0 which is when this update will be released.
* Turn off the grouped settings feature flag to have UI access to all the now-deprecated button type options.
* Open the `wp_options` table in the database (e.g. in phpMyAdmin) and search `option_name` for `woocommerce_woocommerce_payments_version`. We're going to update this (and just this - not the plugin header) while testing.
    * This field holds the previous WCPay version so WCPay can keep track of when it's updated. By setting it to a value lower than the current version, we can simulate an update happening.
    * If the version is lower than `2.6.0`, the migration script will not (and should not!) run so make sure you update the version before each test.
* Visit any wp-admin page. If your payment request button type is set to any of the following (and only the following), the value should be mapped to another value:
    * `Button type: Branded` with `Branded button format: Logo only` should be updated to `Button type: Default`.
    * `Button type: Branded` with `Branded button format: Text and logo` should be updated to `Button type: Buy`.
    * `Button type: Custom` should be updated to `Button type: Buy`.
* After any visit to wp-admin the version number in the database will be updated so, after setting up the next test scenario, make sure to set the version number to something lower than `2.6.0`.

-------------------
